fbclock/stats.go

```diff
@@ -0,0 +1,55 @@
+package fbclock
+
+import "C"
+
+import (
+	"time"
+)
+
+// Stats aggregate stats for fbclock GetTime results
+type Stats struct {
+	Requests    int64
+	Errors      int64
+	WOUAvg      int64
+	WOUMax      int64
+	WOUlt10us   int64
+	WOUlt100us  int64
+	WOUlt1000us int64
+	WOUge1000us int64
+}
+
+// StatsCollector collects stats based on GetTime results
+type StatsCollector struct {
+	stats Stats
+	// internal stuff
+	wouSum int64
+}
+
+// Update processes result of GetTime call and updates Stats
+func (s *StatsCollector) Update(tt *TrueTime, err error) {
+	s.stats.Requests++
+	if err != nil {
+		s.stats.Errors++
+		return
+	}
+	wou := tt.Latest.Sub(tt.Earliest)
+	s.wouSum += int64(wou)
+	s.stats.WOUAvg = s.wouSum / (s.stats.Requests - s.stats.Errors)
+	if wou < 10*time.Microsecond {
```

Comment:
Do you mind adding `<1us` as well?

Reply:
This will make it incompatible with our internal CPP wrapper bucketing, so no.